standalone/inc/pup/PUPScreen.cpp

```diff
@@ -459,15 +459,30 @@ void PUPScreen::QueueBG(int mode)
 
 void PUPScreen::QueueTrigger(char type, int number, int value)
 {
-   if (value == 0)
-      return;
+   // we store the whole state to be able to mach later
+   const std::string name = std::string(1, type) + std::to_string(number);
+   m_triggersState[name] = value;
 
```

Comment:
core/stdafx.h namespaces std::string and std::vector, so you don't need std:: anywhere you use these.